static/js/click_handlers.js

```diff
@@ -501,14 +501,7 @@ $(function () {
         channel.del({
             url: "/json/default_streams" + "?" + $.param({ stream_name: stream_name }),
             error: function (xhr) {
-                var button = row.find("button");
-                if (xhr.status.toString().charAt(0) === "4") {
-                    button.closest("td").html(
-                        $("<p>").addClass("text-error").text(JSON.parse(xhr.responseText).msg)
-                    );
-                } else {
-                    button.text(i18n.t("Failed!"));
-                }
+                ui_report.generic_row_button_error(xhr, $(this));
```

Comment:
We generally try to avoid using `this` over using explicit variables; I think this should be `row` instead?

Reply:
Ohh, yeah, actually I've used it because of use of `this` in above row variable in this handler, anyway I'll change it to `e.target` because. BTW  I've used `btn` as a variable rather than row because of the reason specified in commit message about which I want feedback.

Reply:
Ahh, yeah, totally makes sense -- I think your API is correct (and using `e.target` sounds great).